modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Monoid.kt

```diff
@@ -12,12 +9,12 @@ interface Monoid<A> : Semigroup<A>, TC {
     /**
      * Combine a collection of [A] values.
      */
-    fun combineAll(elems: Collection<A>): A =
-            if (elems.isEmpty()) empty() else elems.reduce { a, b -> combine(a, b) }
+    fun Collection<A>.combineAll(): A =
```

Comment:
Any reason for `Collection` over `Iterable`? `reduce` is an extension function on `Iterable`.

Reply:
It doesn't have isEmpty tho :)

Reply:
>    if (!iterator.hasNext()) throw UnsupportedOperationException("Empty collection can't be reduced.")